SCons/Tool/msvc.xml

```diff
@@ -842,34 +865,42 @@ Specify the location of <filename>vswhere.exe</filename>.
 
 <para>
     The <filename>vswhere.exe</filename> executable is distributed with Microsoft Visual Studio and Build
-    Tools since the 2017 edition, but is also available standalone.
+    Tools since the 2017 edition, but is also available as a standalone installation.
     It provides full information about installations of 2017 and later editions.
     With the <option>-legacy</option> argument, <filename>vswhere.exe</filename> can detect installations of the 2010 through 2015
     editions with limited data returned.
-If <envar>VSWHERE</envar> is set, &SCons; will use that location.
 </para>
 <para>
-    Otherwise &SCons; will look in the following locations and set <envar>VSWHERE</envar> to the path of the first <filename>vswhere.exe</filename>
-located.
+    If &cv-VSWHERE; is set to a <filename>vswhere.exe</filename> location, &SCons; will use that location.
+    When &cv-VSWHERE; is undefined, &SCons; will look in the following locations and set &cv-VSWHERE; to the path
+    of the first <filename>vswhere.exe</filename> located:
 </para>
 
 <itemizedlist>
 <listitem><para><literal>%ProgramFiles(x86)%\Microsoft Visual Studio\Installer</literal></para></listitem>
 <listitem><para><literal>%ProgramFiles%\Microsoft Visual Studio\Installer</literal></para></listitem>
 <listitem><para><literal>%ChocolateyInstall%\bin</literal></para></listitem>
+<listitem><para><literal>%LOCALAPPDATA%\Microsoft\WinGet\Links</literal></para></listitem>
+<listitem><para><literal>~\scoop\shims</literal></para></listitem>
+<listitem><para><literal>%SCOOP%\shims</literal></para></listitem>
 </itemizedlist>
 
 <para>
-  Note that <envar>VSWHERE</envar> must be set at the same time or prior to any of &t-link-msvc;, &t-link-msvs; , and/or &t-link-mslink; &f-link-Tool; being initialized.
+  Note that &cv-VSWHERE; must be set prior to the initial &MSVC; compiler discovery.
+  For example, &cv-VSWHERE; must be set at the same time or before the first msvc &f-link-Tool;
+  (e.g., &t-link-msvc;) or msvc-dependent &f-link-Tool; (e.g., &t-link-midl;) is initialized.
+</para>
 
-  Either set it as follows
+<para>
+  Either set it as follows:
 <programlisting>
 env = Environment(VSWHERE='c:/my/path/to/vswhere')
 </programlisting>
 
-or if your &consenv; is created specifying an empty tools list
-(or a list of tools which omits all of default, msvs, msvc, and mslink),
-and also before &f-link-env-Tool; is called to ininitialize any of those tools:
+Or, if your &consenv; is created specifying: (a) an empty tools list, or (b)
```

Comment:
@mwichmann  - can you wordsmith this sentence a little bit?

Maybe
`Or, if your &consenv; is created specifying: (a) tools=[], or (b) a list of tools which doesn't include any of the following: default, msvc, msvs, mslink, midl and also before  &f-link-env-Tool; is called to ininitialize any of those tools:`

Reply:
The github browser interface is too broken to review this sensibly. I *will* take a look at the doc (.xml) changes, but I have to figure out a reasonable way to do so - won't be today as I've run out of time.

Reply:
While easy to read over, the most important and *technically precise* statement is:
   ```
   Note that &cv-VSWHERE; must be set prior to the initial &MSVC; compiler discovery.
   ```
   
   ***The initial msvc compiler discovery can be triggered without an environment (i.e., prior to the first msvc tool being loaded/initialized).***

   For example, the function `msvc_query_version_toolset` defined in `SCons/Tool/MSCommon/vc.py` can be used to query if a specific toolset is available *without* an environment and *with* a user-specified vswhere executable location:
   ```
   def msvc_query_version_toolset(version=None, prefer_newest: bool=True, vswhere_exe=None):
       """
       Return an msvc version and a toolset version given a version
       specification.

       This is an EXPERIMENTAL proxy for using a toolset version to perform
       msvc instance selection.  This function will be removed when
       toolset version is taken into account during msvc instance selection.

       This function searches for an installed Visual Studio instance that
       contains the requested version. A component suffix (e.g., Exp) is not
       supported for toolset version specifications (e.g., 14.39).

       An MSVCArgumentError is raised when argument validation fails. An
       MSVCToolsetVersionNotFound exception is raised when the requested
       version is not found.

       For Visual Studio 2015 and earlier, the msvc version is returned and
       the toolset version is None.  For Visual Studio 2017 and later, the
       selected toolset version is returned.

       Args:

           version: str
               The version specification may be an msvc version or a toolset
               version.

           prefer_newest: bool
               True:  prefer newer Visual Studio instances.
               False: prefer the "native" Visual Studio instance first. If
                      the native Visual Studio instance is not detected, prefer
                      newer Visual Studio instances.

           vswhere_exe: str
               A vswhere executable location or None.

       Returns:
           tuple: A tuple containing the msvc version and the msvc toolset version.
                  The msvc toolset version may be None.

       Raises:
           MSVCToolsetVersionNotFound: when the specified version is not found.
           MSVCArgumentError: when argument validation fails.
       """
   ```
   
   This PR enforces that only *one* vswhere location is used for detection.  The vswhere location may be explicitly specified by the user (via direct function argument or via the VSWHERE construction variable) or the default location as configured.
   
   The vswhere location must be specified before *first use* which may not necessarily be due to an msvc tool being loaded.

Reply:
While having that bit correct, if the rest is somewhat confusing, clarifying that bit is also critical.  Thus my request for our resident wordsmith to take a pass at it.